Client/Frontend/Browser/HomePageHelper.swift

```diff
@@ -73,4 +73,41 @@ class HomePageHelper {
             })
         navigationController?.presentViewController(alertController, animated: true, completion: nil)
     }
+}
+
+/// Accessors for homepage details from the app state.
+/// These are pure functions, so it's quite ok to have them 
+/// as static.
+class HomePageAccessors {
+    private static let getPrefs = Accessors.getPrefs
+
+    static func getHomePage(state: AppState) -> NSURL? {
+        return getHomePage(getPrefs(state))
+    }
+
+    static func hasHomePage(state: AppState) -> Bool {
+        return getHomePage(state) != nil
+    }
+
+    static func isButtonInMenu(state: AppState) -> Bool {
+        return isButtonInMenu(getPrefs(state))
+    }
+}
+
+private extension HomePageAccessors {
+    static func isButtonInMenu(prefs: Prefs) -> Bool {
+        return prefs.boolForKey(HomePageConstants.HomePageButtonIsInMenuPrefKey) ?? true
+    }
+
+    static func getHomePage(prefs: Prefs) -> NSURL? {
+        let string = prefs.stringForKey(HomePageConstants.HomePageURLPrefKey) ?? getDefaultHomePageString(prefs)
+        guard let urlString = string else {
```

Comment:
this could be combined into one clause:

```
guard let urlString = (prefs.stringForKey(HomePageConstants.HomePageURLPrefKey) ?? getDefaultHomePageString(prefs)) else {
    return nil
}
```